packages/react-badge/src/components/CounterBadge/useCounterBadgeStyles.ts

```diff
@@ -0,0 +1,73 @@
+import { makeStyles, ax } from '@fluentui/react-make-styles';
+import { CounterBadgeState } from './CounterBadge.types';
+import { useRootStyles, useIconStyles } from '../Badge/useBadgeStyles';
+
+/**
+ * Styles for the root slot
+ */
+export const useCounterBadgeRootStyles = makeStyles<CounterBadgeState>([
+  [
+    null,
+    {
+      minWidth: 'auto',
+    },
+  ],
+  [
+    s => s.color === 'warning',
+    theme => ({
+      backgroundColor: theme.global.palette.cranberry.primary,
+      borderColor: theme.global.palette.cranberry.primary,
+    }),
+  ],
+  [
+    s => s.color === 'important',
+    theme => ({
+      backgroundColor: theme.global.palette.grey[14],
+      borderColor: theme.global.palette.grey[14],
+    }),
+  ],
+  [
+    s => s.color === 'severe',
+    theme => ({
+      // @TODO: update these colors once the color used in the design spec has existing color token
+      backgroundColor: theme.global.palette.red.primary,
+      borderColor: theme.global.palette.red.primary,
+    }),
+  ],
+  [
+    s => s.color === 'informative',
+    theme => ({
+      backgroundColor: theme.global.palette.grey[92],
+      borderColor: theme.global.palette.grey[92],
+      color: theme.alias.color.neutral.neutralForeground3,
+    }),
+  ],
+  [
+    s => s.dot,
+    {
+      width: '6px',
+      height: '6px',
+      padding: '0',
+    },
+  ],
+  [
+    s => !s.showZero && s.count === 0 && !s.dot,
+    {
+      display: 'none',
+    },
+  ],
+]);
+
+/**
+ * Applies style classnames to slots
+ */
+export const useCounterBadgeStyles = (state: CounterBadgeState) => {
+  state.className = ax(useRootStyles(state), useCounterBadgeRootStyles(state), state.className);
+  const iconClassName = useIconStyles(state);
+
+  if (state.icon) {
+    state.icon.className = ax(iconClassName, state.icon.className);
+  }
+
+  return state;
+};
```

Comment:
I mentioned this in another comment, but just to clarify, this should be calling `useBadgeStyles`. That'll handle badge's root styles, and the icon slot (and any future slots we might add to the base badge):

```suggestion
export const useCounterBadgeStyles = (state: CounterBadgeState) => {
  state.className = ax(useCounterBadgeRootStyles(state), state.className);
  return useBadgeStyles(state);
};
```